allure-ruby-commons/lib/allure_ruby_commons/result_utils.rb

```diff
@@ -159,7 +159,8 @@ def status(exception)
       # @param [Exception] exception
       # @return [Allure::StatusDetails]
       def status_details(exception)
-        StatusDetails.new(message: exception&.message, trace: exception&.backtrace&.join("\n"))
+        message = exception&.message.nil? ? exception.class.name : exception.message
+        StatusDetails.new(message: message, trace: exception&.backtrace&.join("\n"))
```

Comment:
This can be simplified like so:

```suggestion
        StatusDetails.new(message: exception&.message || exception.class.name, trace: exception&.backtrace&.join("\n"))
```